### PR TITLE
create endpoint for /api/articles/article_id #14

### PR DESCRIPTION
**Pull request by MrChris-H**

### CODE REVIEW

**Review by maddierosecodes — APPROVED** [review]
Looking good.

I don't think the index.js to switch between data types is necessary. You can just require the test data directly into the test suite, this is more efficient as the index.js is performing additional conditional checks :)

**Comment by MrChris-H** [comment]
one of those commits removed the index as requested before

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                created_at: "2020-10-16T05:03:00.000Z",
+                votes: 0,
+              })
+            );
+          });
+      });
+      it('Status 400, msg: "article does not exist"', () => {
```
Comment (emilyb93) [inline __tests__/app.test.js:59]: I would say this for this instance, a 404 would be better as the request is fine, but the article number doesn't exist yet but could do in future. 400 Bad Request is more reserved for bad data type input

File: controllers/article-controllers.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  const { article_id } = req.params;
+  fetchArticle(article_id)
+    .then((article) => {
+      res.status(200).send({ article });
+    })
+    .catch((err) => {
+      next(err);
+    });
+};
+
```
Comment (emilyb93) [inline controllers/article-controllers.js:13]: I would remove this patch request function from this branch and keep each branch as a method and endpoint, as opposed to a branch for an endpoint and all its associated methods.

File: controllers/custom-error-controller.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+exports.customErr = (err, req, res, next) => {
+  if (err.status === 400 && err.msg === "article does not exist") {
```
Comment (emilyb93) [inline controllers/custom-error-controller.js:2]: Again, a 404 would be better for this error as 400 implies Bad Request === bad data input/wrong data type

File: models/articles-models.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+const db = require("../db/connection");
+
+exports.fetchArticle = (id) => {
+  return db
+    .query("SELECT * FROM articles WHERE article_id = $1", [id])
```
Comment (emilyb93) [inline models/articles-models.js:5]: Close your psql strings with a ;

File: models/topics-models.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+const db = require("../db/connection");
+
+exports.fetchTopics = () => {
+  return db.query("SELECT * FROM topics").then(({ rows }) => {
```
Comment (emilyb93) [inline models/topics-models.js:4]: Close your psql strings with a ;

**Review by emilyb93 — CHANGES_REQUESTED** [review]

**Pull request merged**